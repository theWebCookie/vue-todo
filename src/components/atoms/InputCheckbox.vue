<template>
  <input type="checkbox" :disabled="disabled" :class="$style.checkBox" @change="toggleTask(id)" :checked="checked" />
</template>

<script>
export default {
  name: 'InputCheckbox',
  props: {
    disabled: {
      type: Boolean,
      default: false,
    },
    checked: {
      type: Boolean,
      default: false,
    },
    id: {
      type: Number,
    },
  },
  methods: {
    toggleTask(taskId) {
      this.$store.dispatch('toggleTask', taskId);
    },
    removeTask(taskId) {
      this.$store.dispatch('removeTask', taskId);
    },
  },
};
</script>

<style module lang="scss">
.checkBox {
  margin-left: $spacing-small;
}
</style>
