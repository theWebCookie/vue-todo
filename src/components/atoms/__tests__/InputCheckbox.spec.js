import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import { describe, it, expect, jest, beforeEach } from '@jest/globals';
import InputCheckbox from '../InputCheckbox.vue';

describe('InputCheckbox.vue', () => {
  let store;

  beforeEach(() => {
    store = createStore({
      actions: {
        toggleTask: jest.fn(),
      },
    });
  });

  it('renders correctly and matches the snapshot', () => {
    const wrapper = shallowMount(InputCheckbox, {
      global: {
        plugins: [store],
        mocks: {
          $style: {
            label: 'label-class',
            checkBox: 'checkBox-class',
            checked: 'checked-class',
          },
        },
      },
      props: {
        checked: false,
        disabled: false,
        id: 1,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders with checked state', () => {
    const wrapper = shallowMount(InputCheckbox, {
      global: {
        plugins: [store],
        mocks: {
          $style: {
            label: 'label-class',
            checkBox: 'checkBox-class',
            checked: 'checked-class',
          },
        },
      },
      props: {
        checked: true,
        disabled: false,
        id: 1,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders with disabled state', () => {
    const wrapper = shallowMount(InputCheckbox, {
      global: {
        plugins: [store],
        mocks: {
          $style: {
            label: 'label-class',
            checkBox: 'checkBox-class',
            checked: 'checked-class',
          },
        },
      },
      props: {
        checked: false,
        disabled: true,
        id: 1,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('emits change event when toggleTask is called', () => {
    const wrapper = shallowMount(InputCheckbox, {
      global: {
        plugins: [store],
        mocks: {
          $style: {
            label: 'label-class',
            checkBox: 'checkBox-class',
            checked: 'checked-class',
          },
        },
      },
      props: {
        checked: false,
        disabled: true,
        id: 1,
      },
    });

    const taskId = 1;
    wrapper.vm.toggleTask(taskId);
    expect(wrapper.emitted().change).toBeTruthy();
    expect(wrapper.emitted().change[0]).toEqual([taskId]);
  });
});
