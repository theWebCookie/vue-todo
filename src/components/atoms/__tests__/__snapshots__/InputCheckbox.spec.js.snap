// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCheckbox.vue renders correctly and matches the snapshot 1`] = `
<input
  class="checkBox"
  type="checkbox"
/>
`;

exports[`InputCheckbox.vue renders with checked state 1`] = `
<input
  checked=""
  class="checkBox"
  type="checkbox"
/>
`;

exports[`InputCheckbox.vue renders with disabled state 1`] = `
<input
  class="checkBox"
  disabled=""
  type="checkbox"
/>
`;
